src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -199,6 +199,30 @@ public Assembly LoadFromAssemblyName(AssemblyName assemblyName)
             return assembly;
 
         }
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        [SuppressUnmanagedCodeSecurity]
+        private static extern IntPtr LoadUnmanagedLibraryFromPath(string unmanagedDllPath);
+
+        // This method provides a way for overriders of LoadUnmanagedDll() to load an unmanaged DLL from a specific path in a
+        // platform-independent way. The DLL is loaded with default load flags.
+        protected IntPtr LoadUnmanagedDllFromPath(string unmanagedDllPath)
+        {
+            if (unmanagedDllPath == null)
+            {
+                throw new ArgumentNullException("unmanagedDllPath");
+            }
+            if (unmanagedDllPath.Length == 0)
+            {
+                throw new ArgumentException(Environment.GetResourceString("Argument_EmptyPath"), "unmanagedDllPath");
+            }
+            if (Path.IsRelative(unmanagedDllPath))
```

Comment:
If we allow relative paths here and use the system default search order, the behavior would differ from the default search order in `NDirect::LoadLibraryModule`. I figured it would be confusing to have a different default behavior. In the future, if we were to add something like `LoadUnmanagedDllFromName`, it would make sense for such an API to follow what `LoadLibraryModule` does. And if we allow relative paths in this API, it could be confusing to have different behavior between the two for relative paths. My feeling is that we should leave the relative path stuff to a different function and limit this function to the minimum it needs to do, and we can always expand its behavior later if we need it to do more. Thoughts?